disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -35,6 +39,13 @@ maybeGen g = sized $ \s ->
     (1, return Nothing),
     (s, Just <$> resize (s `div` 2) g)]
 
+-- | Wait for a generated `Just` value
+--
+-- Use _only_ in case of emergencies when you have no other way to get an `a` safely
+genFromMaybe :: Gen (Maybe a) -> Gen a
```

Comment:
With this code it will never happen. Instead it will go into infinite loop with the code like this:
```
sample (genFromMaybe (return Nothing :: Gen (Maybe ())))
```
`suchThatMaybe` could be used here instead but it could then fail depending of the `size`.